internal/parser/scanner/states.go

```diff
@@ -356,231 +406,45 @@ var keywordsWithW map[string]token.Type = map[string]token.Type{
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanFKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithF[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithF[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanGKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithG[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithG[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanHKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithH[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithH[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanIKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithI[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithI[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanJKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithJ[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithJ[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanKKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithK[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithK[input], input, s)
+func (s *scanner) scanKeyword() token.Token {
+	nextPos := s.seekNextPos(s.start)
+	input := []rune(strings.ToUpper(string(s.input[s.start:nextPos])))
+	for _, k := range keywordSlice {
+		keyword := []rune(k)
+		j := 0
+		length := len(input)
+		if length > len(keyword) {
+			length = len(keyword)
+		}
+		for i := 0; i < length; i++ {
+			if keyword[i] == input[i] {
+				j++
+			}
+		}
+		if j == len(keyword) {
+			s.acceptString(string(s.input[s.start:nextPos]))
+			return s.createToken(keywordMap[k])
```

Comment:
How is the position pointer moved here?

Reply:
It should be moved to `nextPos`, right? Is this missing? How is this working?

Reply:
`acceptString` internally calls `accept` which moves the pointer

Reply:
Right, of course.